internal/operators/odf/common.go

```diff
@@ -0,0 +1,42 @@
+package odf
+
+import (
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+)
+
+func getODFDeploymentMode(cluster *models.Cluster, odfMinimumHosts int64) odfDeploymentMode {
+	masterHosts, workerHosts, autoAssignHosts := common.GetHostsByEachRole(cluster, true)
+
+	masterCount := len(masterHosts)
+	workerCount := len(workerHosts)
+	autoAssignCount := len(autoAssignHosts)
+	hostCount := masterCount + workerCount + autoAssignCount
+
+	// To keep compatability with the behaviour until now.
+	if hostCount == common.MinMasterHostsNeededForInstallationInHaMode && workerCount == 0 {
+		return compactMode
+	}
+
+	if masterCount == hostCount && masterCount >= int(odfMinimumHosts) {
+		return compactMode
+	}
+
+	if masterCount >= common.MinMasterHostsNeededForInstallationInHaMode && workerCount >= int(odfMinimumHosts) {
+		return standardMode
+	}
+
+	// To be determined, the cluster is not yet in a valid form.
+	return unknown
+}
+
+// To keep compatability with the behaviour until now.
+// TODO - remove this once two control plane nodes OpenShift is implemented
+func isAutoAssignmentAllowed(cluster *models.Cluster) bool {
+	return len(cluster.Hosts) == common.MinMasterHostsNeededForInstallationInHaMode
+}
+
+func willHostPotentiallyRunODF(mode odfDeploymentMode, hostEffectiveRole models.HostRole) bool {
```

Comment:
wouldnt `canRunODF`  or `canHostRunODF` better (naming-wise)? Also maybe we can retrieve `hostEffectiveRole` within this function?

Reply:
The check is not if the host is capable of running the ODF workloads, but will it eventually run them. if it will eventually run them, then we want to verify that it is capable of doing that. We can change it to `willHostRunODF` or something like that

Reply:
WDYT ?

Reply:
What's the difference between `can host eventually run ODF` vs `can host  run ODF`? (honest question)